async-nats/src/auth.rs

```diff
@@ -0,0 +1,24 @@
+// Copyright 2020-2022 The NATS Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#[derive(Clone)]
+pub(crate) enum Authorization {
+    /// No authentication.
+    None,
+
+    /// Authenticate using a token.
+    Token(String),
+
+    /// Authenticate using a username and password.
+    UsernamePassword(String, String),
```

Comment:
Credentials sounds a bit nicer, no?
```suggestion
    Credentials(String, String),
```

Reply:
it does, but we also will have options to load credentials (from .creds file), which would be confusing.